lib/puppet/http/external_client.rb

```diff
@@ -41,8 +41,7 @@ def get(url, headers: {}, params: {}, options: {}, &block)
 
   # (see Puppet::HTTP::Client#post)
   # @api private
-  def post(url, headers: {}, params: {}, options: {}, &block)
-    body = options.fetch(:body) { |_| raise ArgumentError, "'post' requires a 'body' option" }
+  def post(url, body, headers: {}, params: {}, options: {}, &block)
     url = encode_query(url, params)
```

Comment:
It looks like we still need to raise if a body argument isn't given

Reply:
dang, I'll fix that

Reply:
Fixed in https://github.com/puppetlabs/puppet/compare/f571b8b47ae7b271512637619e9e7c5dd0d74d83..6558bd47581199a0e18b72a8b76099108fa26c74